core/constraints/ens.py

```diff
@@ -10,11 +9,10 @@ def __init__(self, user_profile) -> None:
         super().__init__(user_profile)
 
     def is_observed(self, *args, **kwargs) -> bool:
-        from core.models import NetworkTypes
+        from authentication.models import ENSConnection
 
-        user_wallets = self.user_profile.wallets.filter(wallet_type=NetworkTypes.EVM)
-        ens_util = ENSUtil()
-        for wallet in user_wallets:
-            if ens_util.get_name(wallet.address) is not None:
-                return True
-        return False
+        try:
+            ens = ENSConnection.objects.get(user_profile=self.user_profile)
+        except ENSConnection.DoesNotExist:
+            return False
+        return ens.name is not None
```

Comment:
you could use is_connected method instead of doing this, also for getting connection from user_profile we have method in BaseThirdPartyConnection